src/testing/ColdObservable.ts

```diff
@@ -0,0 +1,42 @@
+import Observable from '../Observable';
+import Subscription from '../Subscription';
+import Scheduler from '../Scheduler';
+import TestMessage from './TestMessage';
+import SubscriptionLog from './SubscriptionLog';
+import SubscriptionLoggable from './SubscriptionLoggable';
+import applyMixins from '../util/applyMixins';
+
+export default class ColdObservable<T> extends Observable<T> implements SubscriptionLoggable {
+  public subscriptions: SubscriptionLog[] = [];
+  scheduler: Scheduler;
+  logSubscribedFrame: () => number;
+  logUnsubscribedFrame: (index: number) => void;
+
+  constructor(private messages: TestMessage[],
+              scheduler: Scheduler) {
+    super(function (subscriber) {
+      const observable: ColdObservable<T> = this;
+      const index = observable.logSubscribedFrame();
+      subscriber.add(new Subscription(() => {
+        observable.logUnsubscribedFrame(index);
+      }));
+      observable.scheduleMessages(subscriber);
+      return subscriber;
+    });
+    this.scheduler = scheduler;
+  }
```

Comment:
Above, logSubscribedFrame and logUnsubscribedFrame come from SubscriptionLoggable mixin.